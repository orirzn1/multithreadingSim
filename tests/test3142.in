L15
S7
O5
N14

T0
I@0x00000000
LOAD $2, $0, $6
SUB $4, $0, $6
ADDI $2, $7, 0x1
SUBI $7, $1, 0x4
SUB $5, $0, $4
STORE $4, $6, $3
SUB $4, $7, $1
ADD $3, $1, $1
HALT $0

T1
I@0x00000000
ADD $4, $3, $0
SUBI $7, $5, 0x1
SUBI $2, $0, 0x2
LOAD $3, $0, 0x3
STORE $1, $7, 0x1
HALT $0

T2
I@0x00000000
STORE $3, $4, $5
ADDI $5, $0, 0x2
SUB $3, $0, $2
ADD $1, $7, $4
STORE $0, $2, $0
LOAD $2, $5, 0x4
SUB $6, $1, $0
LOAD $0, $7, 0x2
ADDI $7, $3, 0x1
HALT $0

T3
I@0x00000000
ADD $4, $1, $5
ADDI $7, $0, 0x1
ADDI $2, $3, 0x4
ADD $0, $0, $6
LOAD $4, $5, 0x3
STORE $5, $0, 0x0
LOAD $0, $7, $3
SUB $7, $1, $7
HALT $0

T4
I@0x00000000
STORE $4, $2, $3
ADD $4, $4, $6
ADDI $6, $4, 0x1
SUBI $2, $0, 0x1
SUB $6, $2, $5
ADDI $7, $7, 0x4
LOAD $6, $7, 0x1
LOAD $5, $0, 0x4
HALT $0

T5
I@0x00000000
STORE $5, $0, $2
LOAD $6, $2, $4
SUB $6, $3, $7
ADD $0, $3, $5
ADD $2, $7, $6
HALT $0

T6
I@0x00000000
ADD $1, $0, $5
SUBI $7, $1, 0x1
SUBI $6, $3, 0x1
STORE $7, $2, $5
ADDI $0, $5, 0x2
SUB $2, $4, $0
HALT $0

T7
I@0x00000000
SUB $2, $1, $6
ADDI $6, $7, 0x3
SUB $3, $7, $1
ADD $5, $4, $3
LOAD $3, $0, 0x2
ADDI $2, $4, 0x2
SUBI $2, $2, 0x1
ADD $5, $1, $0
HALT $0

T8
I@0x00000000
SUBI $1, $4, 0x2
LOAD $5, $3, $6
STORE $5, $7, 0x3
ADDI $6, $1, 0x1
SUBI $6, $7, 0x4
STORE $0, $2, 0x0
ADDI $0, $7, 0x1
LOAD $7, $2, $4
SUBI $6, $7, 0x4
SUB $0, $6, $3
HALT $0

T9
I@0x00000000
ADD $2, $0, $3
STORE $0, $3, $1
LOAD $2, $4, 0x0
STORE $3, $6, $7
STORE $2, $3, $4
LOAD $0, $1, 0x1
STORE $5, $0, $2
STORE $3, $3, 0x3
HALT $0

T10
I@0x00000000
ADD $4, $7, $7
ADDI $5, $6, 0x1
SUBI $1, $6, 0x3
SUB $6, $6, $5
ADDI $1, $4, 0x1
ADDI $2, $1, 0x3
HALT $0

T11
I@0x00000000
ADDI $3, $4, 0x1
SUB $5, $1, $1
ADDI $3, $3, 0x2
SUBI $0, $6, 0x2
LOAD $0, $3, $1
LOAD $2, $3, 0x4
HALT $0

T12
I@0x00000000
SUB $5, $2, $5
LOAD $5, $0, $0
SUBI $6, $1, 0x1
LOAD $3, $6, 0x2
SUBI $7, $6, 0x3
SUBI $1, $6, 0x3
HALT $0

T13
I@0x00000000
ADDI $7, $2, 0x2
ADDI $7, $1, 0x3
ADDI $4, $4, 0x1
SUB $0, $0, $1
STORE $7, $5, 0x0
STORE $4, $3, 0x0
STORE $6, $7, $7
SUBI $2, $5, 0x2
ADDI $7, $4, 0x3
HALT $0

D@0x0
0x9
0x3
0x8
0x6
0x1
0x3
0x3
0x2
