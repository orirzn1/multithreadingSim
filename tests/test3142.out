
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0x1	R3 = 0x0	R4 = 0xFFFFFFFC	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFC
Register file thread id 1:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFE	R3 = 0x9	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFF
Register file thread id 2:
	R0 = 0x3	R1 = 0x0	R2 = 0x3	R3 = 0x0	R4 = 0x0	R5 = 0x2	R6 = 0x0	R7 = 0x1
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0x4	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFF
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFF	R3 = 0x0	R4 = 0x0	R5 = 0x3	R6 = 0x3	R7 = 0x4
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 6:
	R0 = 0x2	R1 = 0x0	R2 = 0xFFFFFFFE	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFF	R7 = 0xFFFFFFFF
Register file thread id 7:
	R0 = 0x0	R1 = 0x0	R2 = 0x1	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x3	R7 = 0x0
Register file thread id 8:
	R0 = 0xFFFFFFFF	R1 = 0xFFFFFFFE	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFF	R7 = 0x3
Register file thread id 9:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 10:
	R0 = 0x0	R1 = 0x1	R2 = 0x4	R3 = 0x0	R4 = 0x0	R5 = 0x1	R6 = 0xFFFFFFFF	R7 = 0x0
Register file thread id 11:
	R0 = 0x0	R1 = 0x0	R2 = 0x3	R3 = 0x3	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 12:
	R0 = 0x0	R1 = 0xFFFFFFFC	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFF	R7 = 0xFFFFFFFC
Register file thread id 13:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFE	R3 = 0x0	R4 = 0x1	R5 = 0x0	R6 = 0x0	R7 = 0x4
Blocked MT CPI for this program 3.232759

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0x1	R3 = 0x0	R4 = 0xFFFFFFFC	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFC
Register file thread id 1:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFE	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFF
Register file thread id 2:
	R0 = 0x3	R1 = 0x0	R2 = 0x3	R3 = 0x0	R4 = 0x0	R5 = 0x2	R6 = 0x0	R7 = 0x1
Register file thread id 3:
	R0 = 0x3	R1 = 0x0	R2 = 0x4	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFF
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFF	R3 = 0x0	R4 = 0x0	R5 = 0x3	R6 = 0x3	R7 = 0x4
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 6:
	R0 = 0x2	R1 = 0x0	R2 = 0xFFFFFFFE	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFF	R7 = 0xFFFFFFFF
Register file thread id 7:
	R0 = 0x0	R1 = 0x0	R2 = 0x1	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x3	R7 = 0x0
Register file thread id 8:
	R0 = 0xFFFFFFFF	R1 = 0xFFFFFFFE	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFF	R7 = 0x3
Register file thread id 9:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 10:
	R0 = 0x0	R1 = 0x1	R2 = 0x4	R3 = 0x0	R4 = 0x0	R5 = 0x1	R6 = 0xFFFFFFFF	R7 = 0x0
Register file thread id 11:
	R0 = 0x0	R1 = 0x0	R2 = 0x3	R3 = 0x3	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 12:
	R0 = 0x0	R1 = 0xFFFFFFFC	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFF	R7 = 0xFFFFFFFC
Register file thread id 13:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFE	R3 = 0x0	R4 = 0x1	R5 = 0x0	R6 = 0x0	R7 = 0x4
Finegrained Multithreading CPI for this program 1.008621

